### PR TITLE
ScannerTokens: infer "fewer braces" indent

### PR DESCRIPTION
**Pull request by kitbellew**
For #3045.

### CODE REVIEW

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 1]
```diff
@@ -1609,6 +1599,12 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     }))(location)
   }
 
+  private def isEolAfterColonFewerBracesBody(): Boolean = peekToken match {
```
Comment (tgodzik) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:1602]: We don't check for colon at all here, so this is really just `isNextIndentation`, no?

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala [hunk 2]
```diff
@@ -762,16 +776,37 @@ final class ScannerTokens(val tokens: Tokens)(implicit dialect: Dialect) {
...
+                 *   - after variable and before its type:
+                 *     - within `def`, `val`, `var` declaration or definition: excluded with
+                 *       RegionDefMark
+                 *     - cast within an argument expression: doesn't allow newline as it can then be
+                 *       confused with a fewer-braces invocation
+                 *     - within self-type: not allowed by the compiler
+                 *   - after fewer-braces method call: will apply if not handled otherwise
+                 */
+                def couldBeFewerBraces(): Boolean =
+                  dialect.allowFewerBraces && !sepRegions.contains(RegionDefMark) &&
```
Comment (tgodzik) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala:791]: `!sepRegions.headOption.contains(RegionDefMark) `

Shouldn't it be at the head?

**Review by tgodzik — APPROVED** [review]
Looks good, just some minor comments.

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala (hunk 2 above)
Comment (kitbellew) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala:791]: not really. here's an example:

```
def foo[A : Xxx](bar: Int): Int = 0
```

the first colon and second colon will have a RegionParen/RegionBracket first, before RegionDefMark, but they cannot start fewer braces.

**Pull request merged**